src/EPiServer.SocialAlloy.Web/Social/Repositories/Groups/SocialGroupRepository.cs

```diff
@@ -101,5 +106,90 @@ public SocialGroup Get(string groupName)
 
             return socialGroup;
         }
+
+        /// <summary>
+        /// Retrieves a group based on a list of group ids that are provided.
+        /// </summary>
+        /// <param name="groupIds">The groups ids that are to be retrieved from the underlying data store.</param>
+        /// <returns>The requested groups.</returns>
+        public List<SocialGroup> Get(List<string> groupIds)
+        {
+            List<SocialGroup> socialGroups = new List<SocialGroup>();
+            try
+            {
+                var groupIdList = groupIds.Select(x => GroupId.Create(x)).ToList();
+                var groupCount = groupIdList.Count();
+                var criteria = new CompositeCriteria<GroupFilter, GroupExtensionData>
+                {
+                    Filter = new GroupFilter { GroupIds = groupIdList },
+                    PageInfo = new PageInfo { PageSize= groupCount, CalculateTotalCount= true }
```

Comment:
CalculateTotalCount seems unnecessary here when your page size represents the maximum number of groups that could possibly match.